### PR TITLE
Fix Spelling Errors for endpoint pkg

### PR DESCRIPTION
**Pull request by TrevorTaoARM**
2 spelling errors are fixed which may lead to confusing
for the meaning of the whole sentence and src code.

Signed-off-by: trevor tao <trevor.tao@arm.com>

### CODE REVIEW

File: pkg/endpoint/cache.go [hunk 1]
```diff
@@ -178,7 +178,7 @@ func (ep *epInfoCache) RequireARPPassthrough() bool {
 	return ep.requireARPPassthrough
 }
 
-// RequireEgressProg returns true if the endpoint requires bpf_lxc with esction
+// RequireEgressProg returns true if the endpoint requires bpf_lxc with section
```
Comment (michi-covalent) [inline pkg/endpoint/cache.go]:
```suggestion
// RequireEgressProg returns true if the endpoint requires bpf_lxc with section
```

**Review by michi-covalent — COMMENTED** [review]
thanks for the patch! lgtm other than the trailing space.

**Review by aanm — APPROVED** [review]
Nothing else besides what Michi point it out.

File: pkg/endpoint/cache.go (hunk 1 above)
Comment (TrevorTaoARM) [reply pkg/endpoint/cache.go]: @michi-covalent Updated accordingly, thanks.

**Pull request merged**